--- a/include/typeinfo
+++ b/include/typeinfo
@@ -44,6 +44,7 @@ namespace __cxxabiv1
   class __class_type_info;
 } // namespace __cxxabiv1
 
+#ifndef __GXX_MERGED_TYPEINFO_NAMES
 #if !__GXX_WEAK__
   // If weak symbols are not supported, typeinfo names are not merged.
   #define __GXX_MERGED_TYPEINFO_NAMES 0
@@ -51,6 +52,7 @@ namespace __cxxabiv1
   // On platforms that support weak symbols, typeinfo names are merged.
   #define __GXX_MERGED_TYPEINFO_NAMES 1
 #endif
+#endif
 
 namespace std 
 {
